google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/BatchReadOnlyTransaction.java

```diff
@@ -76,6 +99,30 @@
    *     rows are returned in the natural key order of the index.
    * @param columns the columns to read
    * @param options the options to configure the read
+   *
+   * <!--SNIPPET partition_read_using_index-->
+   * <pre>{@code
+   * final BatchReadOnlyTransaction txn =
+   *     batchClient.batchReadOnlyTransaction(TimestampBound.strong());
+   * List<Partition> partitions =
+   *     txn.partitionReadUsingIndex(
+   *         PartitionOptions.getDefaultInstance(),
+   *         "Singers",
+   *         "SingerId",
+   *         KeySet.all(),
+   *         Arrays.asList("FirstName"));
+   * BatchTransactionId txnID = txn.getBatchTransactionId();
+   * int numRowsRead = 0;
+   * for (Partition p : partitions) {
+   *   BatchReadOnlyTransaction batchTxnOnEachWorker = batchClient.batchReadOnlyTransaction(txnID);
```

Comment:
not sure i understand this example.. its confusing because u are creating a new txn in each iteration of for loop.. not worthy for samples.. 

lets do something simialr to partitionRead

numsRow = unused

Reply:
Thanks, will fix.
I got this e.g. from java-docs-samples but your point is valid.